projects/kit/components/input-files/input-files.component.ts

```diff
@@ -0,0 +1,329 @@
+    }
+
+    get acceptArray(): readonly string[] {
+        return this.getAcceptArray(this.accept);
+    }
+
+    get arrayValue(): ReadonlyArray<TuiFileLike> {
+        return this.getValueArray(this.value);
+    }
+
+    onHovered(hovered: boolean) {
+        this.updateHovered(hovered);
+    }
+
+    onFocused(focused: boolean) {
+        this.updateFocused(focused);
+    }
+
+    onPressed(pressed: boolean) {
+        this.updatePressed(pressed);
+    }
+
+    // TODO: refactor i18n messages
+    onFilesSelected(
+        input: HTMLInputElement,
+        texts: Record<'maxSizeRejectionReason' | 'formatRejectionReason', string>,
+        units: [string, string, string],
+    ) {
+        this.processSelectedFiles(input.files, texts, units);
+        input.value = '';
+    }
+
+    onDropped(
+        event: DataTransfer,
+        texts: Record<'maxSizeRejectionReason' | 'formatRejectionReason', string>,
+        units: [string, string, string],
+    ) {
+        this.processSelectedFiles(event.files, texts, units);
+    }
+
+    onDragOver(dataTransfer: DataTransfer | null) {
+        this.dataTransfer = dataTransfer;
+    }
+
+    removeFile(removedFile: TuiFileLike) {
+        this.updateValue(
+            this.multiple ? this.arrayValue.filter(file => file !== removedFile) : null,
+        );
+    }
+
+    getAppearance(mode: null | unknown): string {
+        return mode === null ? '' : TuiAppearance.Outline;
+    }
+
+    @tuiPure
+    private computeLink$(
+        fileDragged: boolean,
+        multiple: boolean,
+        link: PolymorpheusContent,
+    ): Observable<PolymorpheusContent> {
+        if (fileDragged) {
+            return of('');
+        }
+
+        return this.inputFileTexts$.pipe(
+            map(texts =>
+                multiple && link === ''
+                    ? texts.defaultLinkMultiple
+                    : link || texts.defaultLinkSingle,
+            ),
+        );
+    }
+
+    @tuiPure
+    private computeLabel$(
+        isMobile: boolean,
+        fileDragged: boolean,
+        multiple: boolean,
+        label: PolymorpheusContent,
+    ): Observable<PolymorpheusContent> {
+        if (isMobile) {
+            return of('');
+        }
+
+        if (fileDragged) {
+            return this.inputFileTexts$.pipe(
+                map(texts => (multiple ? texts.dropMultiple : texts.drop)),
+            );
+        }
+
+        return this.inputFileTexts$.pipe(
+            map(texts =>
+                multiple && label === ''
+                    ? texts.defaultLabelMultiple
+                    : label || texts.defaultLabelSingle,
+            ),
+        );
+    }
+
+    @tuiPure
+    private getValueArray(
+        value: TuiFileLike | ReadonlyArray<TuiFileLike> | null,
+    ): ReadonlyArray<TuiFileLike> {
+        if (!value) {
+            return EMPTY_ARRAY;
+        }
+
+        return value instanceof Array ? value : [value];
```

Comment:
`Array.isArray` is [better](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray#instanceof_vs_isarray)